### PR TITLE
bpo-43868: Remove PyOS_ReadlineFunctionPointer from the stable ABI list

### PR DESCRIPTION
**Pull request by encukou**
The inclusion of PyOS_ReadlineFunctionPointer in python3dll.c was a mistake.

According to PEP 384:
> functions expecting FILE* are not part of the ABI, to avoid depending
> on a specific version of the Microsoft C runtime DLL on Windows.

https://bugs.python.org/issue43868

### CODE REVIEW

File: Misc/NEWS.d/next/C API/2021-04-16-18-15-56.bpo-43868.twQ7KH.rst [hunk 1]
```diff
@@ -0,0 +1,3 @@
+:c:func:`PyOS_ReadlineFunctionPointer` is no longer exported by
```
Comment (vstinner) [inline Misc/NEWS.d/next/C API/2021-04-16-18-15-56.bpo-43868.twQ7KH.rst]: Do you think that it's worth it to be mentioned in C API Changes of What's New in Python 3.10?
https://docs.python.org/dev/whatsnew/3.10.html#id2 ?

**Review by vstinner — APPROVED** [review]
LGTM: https://bugs.python.org/issue43868#msg391237

 I just propose to mention the change in What's New in Python 3.10.

File: Misc/NEWS.d/next/C API/2021-04-16-18-15-56.bpo-43868.twQ7KH.rst (hunk 1 above)
Comment (encukou) [reply Misc/NEWS.d/next/C API/2021-04-16-18-15-56.bpo-43868.twQ7KH.rst]: Yes, that sounds reasonable. Thanks for the catch!

**Review by vstinner — APPROVED** [review]
LGTM. Thanks for the What's New entry ;-)

**Pull request merged**